### PR TITLE
Easy fix: Not collected submission counts towards progress bar 

### PR DESCRIPTION
**Pull request by ANUSHIYAPRIYA**
## Motivation and Context
This resolves issue #4502 

## Your Changes

**Description**:
Made changes so that the progress bar in the results page will display [ No.of submissions completed/No of. submissions collected ] ( Ex: 12/15 completed, where 15 - No of submissions collected )

**Type of change** (select all that apply):

 

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

Manually tested in the UI
Added corresponding tests for the changes

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments. 
- [x] I have verified that the TravisCI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [ ] I have updated the Changelog.md file.

### CODE REVIEW

File: app/assets/javascripts/Components/Result/result.jsx [hunk 1]
```diff
@@ -505,6 +505,7 @@ class Result extends React.Component {
           marking_state={this.state.marking_state}
           num_assigned={this.state.num_assigned}
           num_marked={this.state.num_marked}
+          num_collectd={this.props.num_collected}
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Result/result.jsx:508]: - this prop is not used
- a prop named `num_collected` is already being passed to the `SubmissionSelector` on line 499
- `num_collected` should be accessed from the `this.state`, not `this.props` (see other comment below)

File: app/assets/javascripts/Components/Result/result.jsx [hunk 2]
```diff
@@ -505,6 +505,7 @@ class Result extends React.Component {
           marking_state={this.state.marking_state}
           num_assigned={this.state.num_assigned}
```
Comment (mishaschwartz) [inline app/assets/javascripts/Components/Result/result.jsx:506]: - `num_assigned` should no longer be necessary, remove it here and everywhere else

File: app/views/results/edit.html.erb [hunk 3]
```diff
@@ -4,6 +4,7 @@
   <%= javascript_include_tag 'mathjax' %>
   <%= javascript_pack_tag 'result' %>
   <%= javascript_include_tag 'Results/main' %>
+  <% ta_id = @current_user.admin? ? nil : @current_user.id %>
```
Comment (mishaschwartz) [inline app/views/results/edit.html.erb:7]: don't define unnecessary variables in views

File: app/views/results/edit.html.erb [hunk 4]
```diff
@@ -15,6 +16,7 @@
           grouping_id: <%= @result.submission.grouping_id %>,
           assignment_id: <%= @result.submission.grouping.assessment_id %>,
           role: '<%= @current_user.class.name %>',
+          num_collected: <%= @assignment.get_num_collected(ta_id) %>,
```
Comment (mishaschwartz) [inline app/views/results/edit.html.erb:19]: this shouldn't be passed as a prop. Instead, get the information when setting the `state` of the Result component.

**Review by mishaschwartz — CHANGES_REQUESTED** [review]

File: app/controllers/results_controller.rb [hunk 5]
```diff
@@ -154,6 +154,7 @@ def show
           data[:notes_count] = submission.grouping.notes.count
           data[:num_marked] = assignment.get_num_marked(current_user.admin? ? nil : current_user.id)
           data[:num_assigned] = assignment.get_num_assigned(current_user.admin? ? nil : current_user.id)
```
Comment (mishaschwartz) [inline app/controllers/results_controller.rb:156]: If `num_assigned` is not being used please remove it here as well.
If `get_num_assigned` is not being used anywhere else please remove that function as well.

**Review by mishaschwartz — APPROVED** [review]

**Comment by david-yz-liu** [comment]
@ANUSHIYAPRIYA I merged this in because it fixes this specific issue, but it seems like there are other places in the code that still use `get_num_assigned` where it might be more appropriate to use `get_num_collected` instead. So could you please review those places as well?

**Comment by ANUSHIYAPRIYA** [comment]
> @ANUSHIYAPRIYA I merged this in because it fixes this specific issue, but it seems like there are other places in the code that still use `get_num_assigned` where it might be more appropriate to use `get_num_collected` instead. So could you please review those places as well?

Yes, David. I thought of creating a new PR for that.

**Pull request merged**